scripts/ci/test_release_quickstart.py

```
@@ -200,5 +207,375 @@ def test_readmes_do_not_claim_the_quickstart_is_in_cli_archives(self):
+            render_at = step.find("release_readme.py")
+            self.assertGreater(render_at, 0, heading)
+            for name, fragment in PACKED_SOURCE_MEMBERS:
+                copy_at = step.find(fragment)
+                self.assertGreater(copy_at, -1, f"{heading} missing {name}")
+                self.assertLess(copy_at, render_at, f"{heading} {name} copied after render")
+
+    def test_renderer_cli_rejects_arbitrary_archive_path_before_filesystem_access(self):
+        module = load_module()
+        with mock.patch.object(module, "list_archive_members") as list_members:
+            with redirect_stderr(StringIO()):
+                result = module.main(["v5.5.0", "/tmp/attacker-selected-archive"])
```

Ruff (0.16.2)

[error] Probable insecure usage of temporary file or directory: "/tmp/attacker-selected-archive"

(S108)